src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleToAccessProfilesMappingsGenerator.java

```diff
@@ -0,0 +1,125 @@
+
+        Set<String> caseRoles = new HashSet<>();
+        if (!containsCreator) {
+            caseRoles.add(CREATOR.getRole());
+        }
+        caseRoles.addAll(getCaseRolesFromAcls(caseTypeDefinition.getAccessControlLists()));
+        caseRoles.addAll(getCaseRolesFromAcls(eventAcls(caseTypeDefinition)));
+        caseRoles.addAll(getCaseRolesFromAcls(stateAcls(caseTypeDefinition)));
+        caseRoles.addAll(getCaseRolesFromAcls(caseFieldAcls(caseTypeDefinition)));
+        caseRoles.addAll(getCaseRolesFromAcls(caseFieldComplexAcls(caseTypeDefinition)));
+
+        return caseRoles;
+    }
+
+    private List<String> getCaseRolesFromAcls(List<AccessControlList> accessControlLists) {
+        return accessControlLists.stream()
+            .map(AccessControlList::getRole)
+            .filter(role -> role.matches(CASE_ROLE_ID_REGEX))
+            .collect(Collectors.toList());
+    }
+
+    private List<String> getIdamRolesFromAcls(List<AccessControlList> accessControlLists) {
+        return accessControlLists.stream()
+            .map(AccessControlList::getRole)
+            .filter(role -> !role.matches(CASE_ROLE_ID_REGEX))
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessControlList> eventAcls(CaseTypeDefinition caseTypeDefinition) {
+        return caseTypeDefinition.getEvents().stream()
+            .flatMap(element -> element.getAccessControlLists().stream())
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessControlList> stateAcls(CaseTypeDefinition caseTypeDefinition) {
+        return caseTypeDefinition.getStates().stream()
+            .flatMap(element -> element.getAccessControlLists().stream())
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessControlList> caseFieldAcls(CaseTypeDefinition caseTypeDefinition) {
+        return caseTypeDefinition.getCaseFieldDefinitions().stream()
+            .flatMap(element -> element.getAccessControlLists().stream())
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessControlList> caseFieldComplexAcls(CaseTypeDefinition caseTypeDefinition) {
+        return caseTypeDefinition.getCaseFieldDefinitions().stream()
+            .flatMap(element -> element.getComplexACLs().stream())
+            .collect(Collectors.toList());
+    }
+
+    private List<AccessProfile> createAccessProfiles(Set<String> roles, boolean addIdamPrefix) {
+        return roles.stream()
+            .map(role -> createAccessProfile(role, addIdamPrefix))
+            .collect(Collectors.toList());
+    }
+
+    private AccessProfile createAccessProfile(String role, boolean addIdamPrefix) {
+        AccessProfile accessProfile = new AccessProfile();
+        accessProfile.setAccessProfiles(Collections.singletonList(role));
+        accessProfile.setRoleName(addIdamPrefix ? IDAM_PREFIX + role : role);
+        accessProfile.setReadOnly(false);
```

Comment:
At the moment we have in the AccessProfile:
`    private Boolean readOnly;
    private String classification;
    private String roleName;
    private List<String> accessProfiles;`
Ticket mentions `Disabled`, `Authorisation` and others. Should I add them to the AccessProfile?